Improve display of the filter fragment over the search fragment

heiligbasil opened the pull request:
# Description
Move FilterFragment into the view package
Decouple SearchView from fragment to allow filling to the top
Put padding on top of Home/Search fragments to leave room for Search
Enable touch event on the Filter fragment to maintain focus

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)

## Change Status
- [X] Complete, tested, ready to review and merge

# Checklist
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] There are no merge conflicts

heiligbasil committed:
Move FilterFragment into the view package
Decouple SearchView from fragment to allow filling to the top
Put padding on top of Home/Search fragments to leave room for Search
Enable touch event on the Filter fragment to maintain focus

justinbgent reviewed (changes requested):
Interaction with the search bar in the logout fragment is really broken. Need that to be working

heiligbasil committed:
Add attribute in Manifest to prevent bottom controls from rising
Rearrange layout in the Filter Fragment to match new Figma design
Fix Login Fragment layout to allow views to function after change

heiligbasil committed:
Hide keyboard when the Filter Fragment is dismissed

TylerBerrett reviewed (changes requested):
Make the FilterFragment MvvM. This isn't a big enough feature to merge into master

heiligbasil committed:
Add permission in the manifest for coarse location

markpking2 reviewed (approved):
[Most beautiful code I've ever seen](https://res.cloudinary.com/communitycalendar/image/upload/v1579198729/xmvmrmngfaaj9m3dtpgl.png)
